app/workers/certificates.php

```diff
@@ -108,90 +171,140 @@ public function run(): void
+        - certFile: /storage/certificates/{$domain->get()}/fullchain.pem
+          keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+    
+            if (!\file_put_contents(APP_STORAGE_CONFIG . '/' . $domain->get() . '.yml', $config)) {
+                throw new Exception('Failed to save SSL configuration');
+            }
+        } catch(Exception $err) {
+            $errorType = 'error';
 
-            if(!$certificate) {
-                throw new Exception('Failed saving domain to DB');
+            if(\str_starts_with($err->getMessage(), 'Failed to issue a certificate with message')) {
+                $errorType = 'letsEncryptError';
             }
-        }
 
-        $config =
-"tls:
-  certificates:
-    - certFile: /storage/certificates/{$domain->get()}/fullchain.pem
-      keyFile: /storage/certificates/{$domain->get()}/privkey.pem";
+            if(empty($certificate)) {
+                $certificate = new Document([
+                    'domain' => $domain->get(),
+                    'issueDate' => \time(),
+                    'renewDate' => $renew,
+                    'attempts' => 0,
+                    'result' => $errorType,
+                    'log' => \json_encode([
+                        'exception' => $err->getTraceAsString()
+                    ]),
+                ]);
+    
+                $certificate = $dbForConsole->createDocument('certificates', $certificate);
+            } else {
+                $certificate = new Document(\array_merge($certificateArray, [
+                    'issueDate' => \time(),
+                    'renewDate' => $renew,
+                    'attempts' => ((int) $certificateArray['attempts']) + 1,
+                    'result' => $errorType,
+                    'log' => \json_encode([
+                        'exception' => $err->getTraceAsString()
+                    ]),
+                ]));
+    
+                $certificate = $dbForConsole->updateDocument('certificates', $certificate->getId(), $certificate);
+            }
+    
+            if (!$certificate) {
+                throw new Exception('Failed saving failed certificate to DB');
+            }
 
-        if (!\file_put_contents(APP_STORAGE_CONFIG . '/' . $domain->get() . '.yml', $config)) {
-            throw new Exception('Failed to save SSL configuration');
+            throw $err;
         }
-
-        ResqueScheduler::enqueueAt($renew + $safety, 'v1-certificates', 'CertificatesV1', [
-            'document' => [],
-            'domain' => $domain->get(),
-            'validateTarget' => $validateTarget,
-            'validateCNAME' => $validateCNAME,
-        ]);  // Async task rescheduale
-
-        Authorization::reset();
     }
 
     public function shutdown(): void
     {
     }
 }
+
+// We trigger once, to start recursive renewal loop. 5 second delay to stay safe with startup
+ResqueScheduler::enqueueIn(5, Event::CERTIFICATES_QUEUE_NAME, Event::CERTIFICATES_CLASS_NAME, [
```

Comment:
Not sure why are we re-scheduling renewals again? Also, it's a bit hard to track any changes done inside the try-catch block, I would love to revalidate them with you.